### PR TITLE
Add Consent Module - Setup + Configuration + Unit tests

### PR DESCRIPTION
**Pull request by addb**
## Description
- Add new config constant to set and read default consent
- Create a new Consent Module
- Extract collect consent from config
- Add tests for consent module
- Updated scripts and few other places which needs to be updated when adding a new module

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by yangyansong-adbe — APPROVED** [review]
LGTM

File: code/main/consent/consentModule.brs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            ''' TODO:
+            ''' check if consent is udpated and persisted and use that value
+            ''' if not, get the default value from the configuration
+            ''' if both not available, return 'y'
+            ''' return y|p|n
+
+            return "y"
+        end function,
+
+        _extractConsentFromConfiguration: function() as dynamic
```
[the remark this answers is not in the document]
Comment (addb) [reply code/main/consent/consentModule.brs:50]: Sounds good!

File: code/main/consent/consentModule.brs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            ''' Send the consent update request to Edge
+            ''' Update the collect consent in persistence based on the edge request response
+        end function,
+
+        ''' TODO
+        _getCollectConsent: function() as string
+            ''' TODO:
+            ''' check if consent is udpated and persisted and use that value
+            ''' if not, get the default value from the configuration
+            ''' if both not available, return 'y'
```
[the remark this answers is not in the document]
Comment (addb) [reply code/main/consent/consentModule.brs:44]: Yes, I will be adding/Updating all the TODOs in the next PR.

File: code/main/consent/consentModule.brs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        _saveCollectConsent: function(consentValue as string) as void
+            ''' Save the collect consent in persistence
+            ''' y|p|n
+        end function,
+
+        ''' TODO
+        _getCollectConsentFromPersistence: function() as string
+            ''' Get the collect consent from persistence
+            ''' y|p|n
+        end function,
```
[the remark this answers is not in the document]
Comment (addb) [reply code/main/consent/consentModule.brs:99]: Yes, I will be adding/Updating all the TODOs in the next PR. In this PR I have just added as a skeleton.

File: code/main/consent/consentModule.brs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            return collectConsentValue
+        end function,
+
+        _isValidConsentValue: function(consentValue as dynamic) as boolean
+            if _adb_isEmptyOrInvalidString(consentValue) then
+                return false
+            end if
+
+            return (consentValue = "y" or consentValue = "p" or consentValue = "n")
+        end function,
```
[the remark this answers is not in the document]
Comment (addb) [reply code/main/consent/consentModule.brs:81]: I added a (negative) unit test for this: https://github.com/adobe/aepsdk-roku/pull/167/files#diff-09adab1ea0a86f96be5dc7d70d28e7734c82aa97c4bbd59f1afc640f86bc73d7R129
Are you thinking to add more?

**Review by praveek — APPROVED** [review]

**Review by praveek — APPROVED** [review]

File: code/main/consent/consentModule.brs (hunk 2 above)
[the remark this answers is not in the document]
Comment (addb) [reply code/main/consent/consentModule.brs:44]: Yes I meant that only with this comment "if not, get the default value from the configuration module"

File: code/main/consent/consentModule.brs (hunk 4 above)
[the remark this answers is not in the document]
Comment (emdobrin) [reply code/main/consent/consentModule.brs:82]: we could have a complete set of unit tests per unit/method, and _extractCollectConsentValue can assume the expected behavior and have tests for it's own functionality

File: code/main/consent/consentModule.brs (hunk 1 above)
[the remark this answers is not in the document]
Comment (addb) [reply code/main/consent/consentModule.brs]: Return is dynamic to return invalid/null when valid consent is not found in map.

File: code/main/consent/consentModule.brs (hunk 4 above)
[the remark this answers is not in the document]
Comment (addb) [reply code/main/consent/consentModule.brs:82]: I see what you mean.

**Review by emdobrin — APPROVED** [review]

**Pull request merged**